### PR TITLE
arm: Removed support for CC2650

### PR DESCRIPTION
**Pull request by galak**
The SoC, driver, and board support for the CC2650 and CC2650 Sensortag
aren't currently supported and we are removing them as such.  If anyone
is interesting in supporting this platform we can easily recovery it
from git.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Comment by galak** [comment]
Note: this removal of this was discussed at the Zephyr TSC Meeting on Jan 15th, and voted on by the TSC and the removal was approved.

Anyone interested in supporting this platform is free to revive this, but will need to be willing to update drivers and such to match broader Zephyr changes (like GPIO API rework).

File: doc/releases/release-notes-2.2.rst [hunk 1]
```diff
@@ -76,7 +76,7 @@ Architectures
 
 * ARM:
 
-  * <TBD>
+  * Removed support for CC2650
```
Comment (ioannisg) [inline doc/releases/release-notes-2.2.rst]: Hm, maybe give the full board name, cc2650_sensortag?

**Review by ioannisg — APPROVED** [review]
Fine with me, I only have a comment about the rel-notes.
@galak are we also removing the hal-suport from EXT TI?

**Review by vanti — APPROVED** [review]
LGTM

**Review by mnkp — APPROVED** [review]

**Comment by galak** [comment]
> @galak are we also removing the hal-suport from EXT TI?

need to look at that.

**Pull request merged**